packages/nimble-components/src/rich-text/models/rich-text-tracker.ts

```diff
@@ -0,0 +1,92 @@
+import { DOM } from '@microsoft/fast-element';
+import type { MentionInternals } from '../../rich-text-mention/base/models/mention-internals';
+import { UpdateTracker } from '../../utilities/models/update-tracker';
+import type { RichText } from '../base';
+
+const isMentionInternalsProperty = (
+    changedProperty: string,
+    ...args: (keyof MentionInternals)[]
+): boolean => {
+    for (const arg of args) {
+        if (changedProperty === arg) {
+            return true;
+        }
+    }
+    return false;
+};
+
+const trackedItems = ['pattern', 'mappingConfigs', 'buttonLabel'] as const;
+
+/**
+ * Helper class to track what updates are needed to the rich text components based on configuration
+ * changes.
+ */
+export class RichTextUpdateTracker extends UpdateTracker<typeof trackedItems> {
+    private updateQueued = false;
+
+    public constructor(private readonly richText: RichText) {
+        super(trackedItems);
+    }
+
+    public get updatePattern(): boolean {
+        return this.isTracked('pattern');
+    }
+
+    public get updateMappingConfigs(): boolean {
+        return this.isTracked('mappingConfigs');
+    }
+
+    public get updateButtonLabel(): boolean {
+        return this.isTracked('buttonLabel');
+    }
+
+    public trackMentionInternalsPropertyChanged(
+        changedMentionInternalsProperty: string
+    ): void {
+        if (
+            isMentionInternalsProperty(
```

Comment:
`isMentionInternalsProperty` is only ever called with a single arg, so why not just use a simple `===` test instead? In fact, this can just be a `switch` statement. We can remove `isMentionInternalsProperty`.

Reply:
Updated the changes